src/token_position.ts

```diff
@@ -145,6 +145,54 @@ const VARIABLE_TOKENIZER = /(^\w+)/
 const ASCII_CHARACTER_TOKENIZER = /(^[\x21-\x2F|\x3A-\x40|\x5B-\x60|\x7B-\x7E])/
 const NONVARIABLE_TOKENIZER = /(^[^\x21-\x7E]+)/
 
+const enum TokenType {
+    /** Tokens that are identities, i.e. variable names */
+    Identity,
+    /** Tokens that are ascii characters but aren't in identies (i.e. {, }, [, ], |, ;,  etc) */
+    ASCII,
+    /** Every token we encounter that doesn't fall into the other two TokenTypes */
+    NonVariable,
+}
```

Comment:
Yea, I thought it was `Identity` in the go parser but its [just `Ident`](https://sourcegraph.com/github.com/golang/go@afae876b3189711c49f9d2cb197081f41c9f3979/-/blob/src/go/token/token.go#L25) and they mention identifier in the comments. The name is confusing anyways because it can include keywords too. I'll switch it to `Alphanumeric`.